extensions/functions_arithmetic.yaml

```diff
@@ -1704,28 +1704,6 @@ window_functions:
         decomposable: NONE
         return: i64?
         window_type: PARTITION
-  - name: "first_value"
-    description: >
-      Returns the first value in the window.
-    impls:
-      - args:
-          - value: any1
-            name: expression
-        nullability: DECLARED_OUTPUT
-        decomposable: NONE
-        return: any1
-        window_type: PARTITION
-  - name: "last_value"
-    description: >
-      Returns the last value in the window.
-    impls:
-      - args:
-          - value: any1
-            name: expression
-        nullability: DECLARED_OUTPUT
-        decomposable: NONE
-        return: any1
-        window_type: PARTITION
```

Comment:
Heads up, moving _existing_ functions is a breaking change, and a relatively painful one to workaround. I would prefer if we could avoid breakage here.

See: https://github.com/substrait-io/substrait/issues/634#issuecomment-2091386661

Reply:
I guess the options are:
1. move existing functions (breaking change
2. duplicate existing functions (bad)
3. add new functions with new name (leads to confusing end state as there's now two functions with same functionality but different name, unless we can deprecate the existing functions somehow)

For my need, I think both 1 and 3 work fine, so I don't have strong opinions - I guess it's a question of breaking now vs keeping a worse state for ever/until breaking later?

Reply:
4. We could also keep the functions in the same file, but move them into the aggregate functions block.

> I guess it's a question of breaking now vs keeping a worse state for ever/until breaking later?

I think it would good to move these eventually, but it would be nice to do _after_ substrait-java can handle duplicate functions in different files, because then we could make the change by duplicating the functions into the new file in one release, and then removing the old functions in the next release.


I've filed https://github.com/substrait-io/substrait-java/issues/275 to track this work in substrait-java.

Reply:
Keeping the same file for now, fixing substrait-java, and then moving files works for me.

Reply:
Ah, I thought "moving" meant moving also from window -> aggregation. Keeping them in the old file is fine for me -like fc3fa78a7a1e3c3cf9ba43abdb2cc7fbfc635aae ?